### PR TITLE
Inline all aliasing let bindings generated during conversion to DAML-LF

### PR DESCRIPTION
**Pull request by martin-drhu-da**
Inlining the aliases generated for record wildcards is a prerequisite for
the maintainer clauses of our contract keys syntax to work.

Inlining all other aliases doesn't seem to be a bad idea as well.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](./CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](./docs/source/support/release-notes.rst), if appropriate

### CODE REVIEW

**Comment by shayne-fletcher-da** [comment]
👍

File: daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs [hunk 1]
```diff
@@ -809,31 +805,24 @@ convertExpr env0 e = do
           TContractId{} -> asLet
           TUpdate{} -> asLet
           TScenario{} -> asLet
-          tcon0 -> do
-              ctor <- toCtor env con
+          tcon -> do
+              ctor@(Ctor _ fldNames fldTys) <- toCtor env con
               if not (isRecordCtor ctor)
-                then do
+                then convertLet env bind scrutinee $  \env -> do
```
Comment (neil-da) [inline daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]:
```suggestion
                then convertLet env bind scrutinee $ \env -> do
```

File: daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs [hunk 2]
```diff
@@ -809,31 +805,24 @@ convertExpr env0 e = do
...
+                  bind' <- convertExpr env (Var bind)
                   ty <- convertType env $ varType bind
                   alt' <- convertAlt env ty alt
-                  bind' <- convVarWithType env bind
-                  pure $
-                    ELet (Binding bind' scrutinee') $
-                    ECase (EVar $ convVar bind) [alt']
-                else mkProj ctor tcon0 scrutinee'
+                  pure $ ECase bind' [alt']
+                -- NOTE(MH): The 'zipExact' below fails for pattern matches on constructors with existentials.
```
Comment (neil-da) [inline daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: Comment seems to add zero value, since the same comment is in the error message

Comment (martin-drhu-da) [reply daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: You're right. I removed it.

**Pull request merged**